contrib/opencensus-ext-azure/opencensus/ext/azure/common/__init__.py

```diff
@@ -48,6 +48,13 @@ def process_options(options):
         or 'https://dc.services.visualstudio.com'
     options.endpoint = endpoint + '/v2/track'
 
+    # Authorization
+    # `azure.core.credentials.TokenCredential` class must be valid
+    if options.credential and not hasattr(options.credential, 'get_token'):
```

Comment:
Checking to see if a method exists by name to check type seems odd.  You should be checking for type.  Consider using isinstance(..) https://docs.python.org/2.7/library/functions.html#isinstance

Reply:
Yes type checking is definitely more preferred. However, if you take a look at the actual implementations `TokenCredential` classes in Python Azure SDK, they actually implement them without an interface. There is no base class that is common amongst all types of `TokenCredential` s. This means that if ever a `TokenCredential` is implemented that does NOT implement `get_token` (since in Python, it is not enforced), our code will actually crash if we do not have the `hasattr` check. With `hasattr`, no matter how azure sdk implements the `TokenCredential`, as long as we have the function we need, we are okay.

Take a look  at [ManagedIdentityCredential](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/identity/azure-identity/azure/identity/_credentials/managed_identity.py#L42) as an example. Notice how it only extends `object`.

Also, Python azure sdk only defined [TokenCredential](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/core/azure-core/azure/core/credentials.py#L16) class for type checking (linting). There is no enforcement of actual implementations at all.

Reply:
uh.  That is weird!  Oh well.